tests/unittest_regrtest.py

```diff
@@ -399,5 +401,33 @@ class Another(subclass):
     parse(code)
 
 
+def test_max_inferred_for_complicated_class_hierarchy() -> None:
+    """Regression test for a crash reported in https://github.com/PyCQA/pylint/issues/5679.
+
+    The class hierarchy of 'sqlalchemy' is so intricate that it becomes uninferable with
+    the standard max_inferred of 100. We used to crash when this happened.
+    """
+    # Create module and get relevant nodes
+    module = resources.build_file(
+        str(resources.RESOURCE_PATH / "max_inferable_limit_for_classes" / "main.py")
+    )
+    init_attr_node = module.body[-1].body[0].body[0].value.func
+    init_object_node = module.body[-1].mro()[-1]["__init__"]
+    super_node = next(init_attr_node.expr.infer())
+
+    # Arbitrarily limit the max number of infered nodes per context
+    InferenceContext.max_inferred = -1
+    context = InferenceContext()
+
+    # Try to infer 'object.__init__' > because of limit is impossible
+    for _ in bases._infer_stmts([init_object_node], context, frame=super):
+        pass
```

Comment:
```suggestion
        assert _ == Uninferable
```
```suggestion
        pass
    assert _ == Uninferable
```
I'll let you apply the one you wanted :)

Reply:
Did another change, where `_` is also renamed. If this passes the CI this should be ready for review again!